src/api/emailNotifications/client.ts

```diff
@@ -0,0 +1,70 @@
+import { HttpError } from 'app/../error/AppErrors';
+
+import config from 'app/config';
+
+import createHttpClient from '../client';
+import {
+  SubscribePayload,
+  UnsubscribePayload,
+  AuthenticatePayload,
+  ModifyPayload,
+  CheckSubscriptionPayload,
+  VerifyEmailPayload,
+  VerifyEmailResponse,
+  SubscribeResponse,
+  ModifyResponse,
+  CheckSubscriptionResponse,
+} from './types';
+
+export enum EmailNotificationsError {
+  INVALID_EMAIL = 'Enter a valid email address',
+  ERROR_VERIFICATION_ERROR = 'On /verify-email/ request the server responded with result: "error"',
+  WRONG_PIN = 'Wrong pin',
+  ERROR_REQUEST_AUTHENTICATION = 'On /authenticate/ request the server responded with result: "error',
+}
+
+const httpClient = createHttpClient(config.emailNotificationsApi);
+
+export const subscribeEmail = async (data: SubscribePayload): Promise<SubscribeResponse> => {
```

Comment:
I think that this should be parsed inside HTTP client in onResponde method.  Here we should have
```
export const unsubscribeEmail = (data: UnsubscribePayload): Promise<UnsubscribePayload> => httpClient.post(`/unsubscribe/`, data);
```

Reply:
Yea, but the problem appears when you want to handle some specific responses, like the ones with `"status": "error"` or when an endpoint behaves differently depending on the request. 

If it doesn't convince you I'll change it as you said

Reply:
I see that in every request we use `data` so I would move it to HTTP client

Reply:
Done